version/src/main/java/org/jboss/as/version/ProductConfig.java

```diff
@@ -262,10 +261,10 @@ private static void safeClose(Closeable c) {
 
     private static class ProductConfProps {
         private final Properties miscProperties;
-        private final ModuleIdentifier productModuleId;
+        private final String productModuleId;
 
         private ProductConfProps(String slot) {
-            this.productModuleId = slot == null ? null : ModuleIdentifier.create("org.jboss.as.product", slot);
+            this.productModuleId = slot == null ? null : "org.jboss.as.product:" + slot;
```

Comment:
@yersan This needs to be:

this.productModuleId = slot == null ? null : ("main",equals(slot) ? "org.jboss.as.product" : "org.jboss.as.product:" + slot);

The canonical string representation of a module with slot 'main' is to NOT include ":main".  The ModuleIdentifier class implements that logic. Where we discard its use we need to implement it ourselves.

I want to add utility methods that we can call throughout WF to handle this kind of thing, somewhere in the 'controller' or 'server' modules, but for the 'version' module I think I just needs to be manually coded.

Reply:
#6270is the proposed utility. But, not usable here. Not really needed either, as the use case here is simple. That PR has a lot of code related to escaping arbitrary inputs but I think this code can assume that any slot value is simple text that doesn't need escaping.

Reply:
@bstansberry Done